### PR TITLE
Add mcpu pattern for Morello and related tests in buildbot

### PR DESCRIPTION
**Pull request by drpdr**
Added needed `mpu` argument pattern to support bitcode generation in morello. Removed (no longer needed) test for sqlite and added lua tests for riscv64 and morello-purecap.

### CODE REVIEW

**Comment by drpdr** [comment]
bors try

File: .buildbot.sh [hunk 1]
```diff
@@ -1,31 +1,48 @@
...
-	echo "Only riscv64 is supported."
+ARCH=
+CHERI=
+CONFIG_FLAGS=
+
+if [ "$1" = "morello-purecap" ]; then
+	ARCH="morello"
+	CHERI=$HOME/cheri/output/"$ARCH"-sdk
+	CONFIG_FLAGS="-target aarch64-unknown-freebsd13 \
+	--sysroot="$HOME"/cheri/output/rootfs-morello-purecap \
```
Comment (ltratt) [inline .buildbot.sh]: Need to indent lines follow a `\`.

File: .buildbot.sh [hunk 2]
```diff
@@ -1,31 +1,48 @@
...
+	-B"$HOME"/cheri/output/morello-sdk/bin \
+	-mcpu=rainier -march=morello+c64 -mabi=purecap"
+elif [ "$1" = "riscv64" ]; then
```
Comment (ltratt) [inline .buildbot.sh]: This will also match riscv-64 hybrid I guess?

File: shared/parser.go [hunk 3]
```diff
@@ -351,7 +351,7 @@ func Parse(argList []string) ParserResult {
 		"-fopenmp":       {0, pr.compileLinkUnaryCallback},
 
 		"-Wl,-dead_strip": {0, pr.warningLinkUnaryCallback},
-		"-dead_strip":     {0, pr.warningLinkUnaryCallback}, //iam: tor does this. We lose the bitcode :-(
+		"-dead_strip":     {0, pr.warningLinkUnaryCallback},  //iam: tor does this. We lose the bitcode :-(
```
Comment (ltratt) [inline shared/parser.go]: I don't think the ` ` addition is intentional?

**Comment by drpdr** [comment]
bors try

File: .buildbot.sh (hunk 2 above)
Comment (drpdr) [reply .buildbot.sh]: We do not build riscv64 hybrid images.

File: shared/parser.go (hunk 3 above)
Comment (drpdr) [reply shared/parser.go]: I *think* is related to the `go fmt` (probably a different version?) but I'll try to fix it.

File: .buildbot.sh (hunk 2 above)
Comment (ltratt) [reply .buildbot.sh]: Yes, but we might, so can we check for that here?

File: .buildbot.sh (hunk 1 above)
Comment (drpdr) [reply .buildbot.sh]: Fixed (???) in [2314e76](https://github.com/capablevms/gllvm/pull/3/commits/2314e767e52ee235be3833c14ea1740427d62871). Let me know if I got this wrong, please.

Comment (ltratt) [reply .buildbot.sh]: Too much (and too inconsistent) indent ;) Two spaces would fit in with the rest of the file.

File: shared/parser.go (hunk 3 above)
Comment (drpdr) [reply shared/parser.go]: FIx in [7a7facb](https://github.com/capablevms/gllvm/pull/3/commits/7a7facb04ac073ce43dfbd95f6d72eeabc4b66cb).

File: .buildbot.sh (hunk 2 above)
Comment (drpdr) [reply .buildbot.sh]: Of course.

File: .buildbot.sh (hunk 1 above)
Comment (drpdr) [reply .buildbot.sh]: I am still not sure. In [9fccbc4](https://github.com/capablevms/gllvm/pull/3/commits/9fccbc4ef1bda10122c1b834d45ff721a45db617)?

File: .buildbot.sh (hunk 2 above)
Comment (drpdr) [reply .buildbot.sh]: Fix in [78c46e2](https://github.com/capablevms/gllvm/pull/3/commits/78c46e25af34e18c5f7a203f047ed69c2cd225d6).

File: .buildbot.sh [hunk 4]
```diff
@@ -12,16 +12,16 @@ if [ "$1" = "morello-purecap" ]; then
 	ARCH="morello"
 	CHERI=$HOME/cheri/output/"$ARCH"-sdk
 	CONFIG_FLAGS="-target aarch64-unknown-freebsd13 \
-				  --sysroot="$HOME"/cheri/output/rootfs-morello-purecap \
-	              -B"$HOME"/cheri/output/morello-sdk/bin \
-	              -mcpu=rainier -march=morello+c64 -mabi=purecap"
+	  --sysroot="$HOME"/cheri/output/rootfs-morello-purecap \
```
Comment (ltratt) [inline .buildbot.sh]: The nested `"`s will have some odd effects. Better to use `${HOME}` and remove the quotes.

File: .buildbot.sh (hunk 1 above)
Comment (ltratt) [reply .buildbot.sh]: Fine!

File: .buildbot.sh (hunk 4 above)
Comment (drpdr) [reply .buildbot.sh]: Fix in [8657556](https://github.com/capablevms/gllvm/pull/3/commits/86575565fea7cef54844324f3ce9330d2c728adb).

**Comment by ltratt** [comment]
Please squash while also thinking whether the commit message should say something like "Add -mcpu *flag*" rather than "argument pattern" (I don't know what an "argument pattern" is!).,

**Comment by drpdr** [comment]
> Please squash while also thinking whether the commit message should say something like "Add -mcpu _flag_" rather than "argument pattern" (I don't know what an "argument pattern" is!).,

Sure. I've just repeated what the code says. I'll follow your suggestion, though.

**Comment by drpdr** [comment]
Squashed.

**Pull request merged**